chromadb/segment/impl/vector/grpc_segment.py

```diff
@@ -96,10 +105,18 @@ def query_vectors(
                 query["request_version_context"]
             ),
         )
-        response: QueryVectorsResponse = self._vector_reader_stub.QueryVectors(
-            request,
-            timeout=self._request_timeout_seconds,
-        )
+
+        try:
+            response: QueryVectorsResponse = self._vector_reader_stub.QueryVectors(
+                request,
+                timeout=self._request_timeout_seconds,
+            )
+        except grpc.RpcError as rpc_error:
+            message = rpc_error.details()
+            if "Collection version mismatch" in message:
```

Comment:
unfortunately this is the best non-bloaty way to do this, alternative is PB in detail